### DISCUSSION TITLE
Limit API rules and filters to the current user

### DISCUSSION THREAD

**Discussion by brucewayyn** [pocketbase/pocketbase#2194] [Q&A] [answered]
I want only the current user to view, read, write, and delete their addresses. Is the following accurate:

<img width="691" alt="Screenshot 2023-04-01 at 2 55 56 PM" src="https://user-images.githubusercontent.com/122556307/229284705-07077cc8-6185-44c6-89d8-7f1f0eb3b33e.png">

Similarly, if a user chooses to share only part of their primary address (e.g. country and state fields only), should the `View rule` be set to 'empty`, and handle the conditionals in frontend to view the desired fields?

**Comment by ganigeorgiev** [comment, marked as answer]
> I want only the current user to view, read, write, and delete their addresses. Is the following accurate:

Not exactly. `@request.data.*` refers to the data submitted with the request (usually available only with the create and update actions). If you want to check an already saved record field, you can do that directly: `@request.auth.id = user`.

Assuming that `user` is a **nonempty** single `relation` field, here are example rules:

1. List/Search rule:
   ```js
   @request.auth.id = user
   ```

2. View rule:
   ```js
   @request.auth.id = user
   ```

3. Create rule:
   ```js
   @request.auth.id = @request.data.user

   // this will also work, since for create we populate the submitted request data as record data
   @request.auth.id = user
   ```

4. Update rule:
   ```js
   @request.auth.id = user && 
   (@request.data.user:isset = false || @request.auth.id = @request.data.user)
   ```
   _(the second condition ensures that no new `user` value was submitted; if you want to allow the authenticated user to change the "ownership" of the record and to assign it to some other user, you can remove it)_

5. Delete rule:
   ```js
   @request.auth.id = user
   ```

---

> Similarly, if a user chooses to share only part of their primary address (e.g. country and state fields only), should the View rule be set to 'empty`, and handle the conditionals in frontend to view the desired fields?

I'm not sure I follow. If you want to keep certain user fields private you can move them in a separate collection (eg. `profile`) and have a relation back to the `users` collection. This way you'll have individual API rules and can decide who have access to the `profile` data. You can fetch the relation together with the main response by using [`?expand`](https://pocketbase.io/docs/expanding-relations/).

**Reply by brucewayyn** [reply]
Thank you. I wish there was a way to view the `@relations` between `fields` and `collections` in a visual way.

**Reply by brucewayyn** [reply]
> I'm not sure I follow. If you want to keep certain user fields private you can move them in a separate collection (eg. `profile`) and have a relation back to the `users` collection. This way you'll have individual API rules and can decide who have access to the `profile` data. You can fetch the relation together with the main response by using [`?expand`](https://pocketbase.io/docs/expanding-relations/).

In this case, is `Type: View` recommended for a collection such as `profile`?

**Reply by ganigeorgiev** [reply]
It is hard to give a concrete answer without the full picture of your use case and what you are trying to do.

There are several approaches you can take depending on the access pattern you want to use.

Views are **read-only**, meaning that if you just want to provide read access to a subset of certain public user fields, then Yes, you can for example lock the default list/view collection rules and create a new view collection with only the fields you want to be visible.

**Reply by AndiKod** [reply]
The 'Marked as answer' comment's snippets should be added to the Examples section in the Documentation. The basic "show only current user's content" might be the most expected filter.

**Comment by SardorbekAbdubannopo** [comment]
Write an API to change the data of a single selected article. Make it possible to change only if the owner of the article is the current user

**Comment by SardorbekAbdubannopo** [comment]
Write an API to change the data of a single selected article. Make it possible to change only if the owner of the article is the current user
